From 1b783aec1fe643f02d9615693996f397a2f48261 Mon Sep 17 00:00:00 2001
From: Michael Pauzner <mpauzner@habana.ai>
Date: Tue, 06 May 2025 18:15:53 +0300
Subject: [PATCH] [SW-221270] gaudi3: serdes: clean-up serdes table

Cleanup unused source code under #ifdef section.

Change-Id: I0c0efc746145e87a886495f65bd2ff566d8df926
Signed-off-by: Michael Pauzner <mpauzner@habana.ai>
(cherry picked from commit 4ef4d4435c6b81a7293caf11aef72b95e10779db)
---

diff --git a/include/hl_serdes.h b/include/hl_serdes.h
index c1febe3..432caf9 100644
--- a/include/hl_serdes.h
+++ b/include/hl_serdes.h
@@ -14,7 +14,6 @@
 #define MAX_MODULES_IN_CHASSIS		8
 #define HL_UNKNOWN_SERDES_TYPE		0
 #define HL_SERDES_UNINITIALIZED_TYPE	~0
-#undef SERDES_TABLE_SUPPORT
 
 /* hl_serdes_config is temporary kept before removal */
 struct hl_serdes_config {
@@ -35,11 +34,9 @@
 int hl_serdes_get_serdes_info_core(const struct device *dev, struct cpucp_nic_info *nic_info,
 				   uint16_t serdes_type);
 
-#ifndef SERDES_TABLE_SUPPORT
 int hl_serdes_config_init(uint8_t module_id, uint8_t link_id);
 const struct hl_serdes_module_config *hl_serdes_module_config_get(void);
 const struct hl_serdes_link_id_config *hl_serdes_link_id_config_get(void);
-#endif
 
 uint32_t hl_serdes_get_link_mask(void);
 
diff --git a/src/hl_serdes/gaudi3/serdes.c b/src/hl_serdes/gaudi3/serdes.c
index 8fcc008..747e71a 100644
--- a/src/hl_serdes/gaudi3/serdes.c
+++ b/src/hl_serdes/gaudi3/serdes.c
@@ -37,17 +37,6 @@
 	GAUDI3_HLB325_FULL_OAM_3PORTS_SCALE_OUT,
 };
 
-struct hl_serdes_config_pcie {
-	uint8_t link_config;
-	uint8_t reserved;
-	uint16_t serdes_type;
-	uint32_t link_ext_mask;
-	uint32_t link_mask;
-	uint32_t auto_neg_mask;
-	uint64_t pol_rx_mask;
-	uint64_t pol_tx_mask;
-};
-
 struct hl_serdes_oam_data {
 	uint8_t module_id;
 	uint8_t link_id;
@@ -80,18 +69,6 @@
 
 #define RACK_NODE_ID_UNDEFINED		255
 
-#ifdef SERDES_TABLE_SUPPORT
-static struct hl_serdes_config_pcie cfg_pcie;
-#endif
-
-int hl_serdes_get_polarity_mask(uint64_t *tx_mask, uint64_t *rx_mask)
-{
-	*tx_mask = HL_SERDES_TX_POLARITY;
-	*rx_mask = HL_SERDES_RX_POLARITY;
-
-	return 0;
-}
-
 #define ETHERNET_DEFAULT_PORT_ENABLE	0x00FFFFFF
 
 static uint32_t get_boot_nic_mask(void)
@@ -104,297 +81,6 @@
 	return nic_mask;
 }
 
-#ifdef SERDES_TABLE_SUPPORT
-#if !defined(TARGET_IS_PALAD)
-static int dev_serdes_type = HL_SERDES_UNINITIALIZED_TYPE;
-#endif
-static uint16_t serdes_type;
-
-static struct hl_serdes_config serdes_table[] = {
-	[GAUDI3_UNKNOWN_SERDES_TYPE] = {
-		HLS3_FULLSCALE_IN_SERDES_TYPE,
-		HLS3_UNKNOWN_SERDES_LINK_ID,
-		{0, 0, 0, 0, 0, 0, 0, 0},
-		{0, 0, 0, 0, 0, 0, 0, 0},
-		{0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF,
-		 0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF},
-		{0, 0, 0, 0, 0, 0, 0, 0},
-		{0, 0, 0, 0, 0, 0, 0, 0},
-		{0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
-	},
-	[GAUDI3_FULLSCALE_IN] = {
-		HLS3_FULLSCALE_IN_SERDES_TYPE,
-		HLS3_FULLSCALE_IN_SERDES_LINK_ID,
-		{0x000000000000, 0x000000000000, 0x000000000000, 0x0100E0000000,
-		 0x000000000000, 0x000000008000, 0x00000000D000, 0x000000000000},
-		{0x00000030F010, 0x0C0000000003, 0x0CF20000000F, 0x000000000300,
-		 0x000000001060, 0x00F001F00000, 0x030000F00000, 0x000000000000},
-		{0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF,
-		 0xFFFFFF, 0xFFFFFF},
-		{0, 0, 0, 0, 0, 0, 0, 0},
-		{0x3FFF, 0xFFF003, 0xFFF003, 0x3FFF,
-		 0xC00FFF, 0xFFFC00, 0xFFFC00, 0xC00FFF},
-		{0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
-	},
-	[GAUDI3_FULLSCALE_OUT] = {
-		HLS3_FULLSCALE_OUT_SERDES_TYPE,
-		HLS3_FULLSCALE_OUT_SERDES_LINK_ID,
-		{0x000000000000, 0x000000000000, 0x000000000000, 0x0100E0000000,
-		 0x000000000000, 0x000000008000, 0x00000000D000, 0x000000000000},
-		{0x00000030F010, 0x0C0000000003, 0x0CF20000000F, 0x000000000300,
-		 0x000000001060, 0x00F001F00000, 0x030000F00000, 0x000000000000},
-		{0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF,
-		 0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF},
-		{0xFFC000, 0x000FFC, 0x000FFC, 0xFFC000,
-		 0x3FF000, 0x0003FF, 0x0003FF, 0x3FF000},
-		{0x3FFF, 0xFFF003, 0xFFF003, 0x3FFF,
-		 0xC00FFF, 0xFFFC00, 0xFFFC00, 0xC00FFF},
-		{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
-	},
-	[GAUDI3_FULLOAM_3PORTS_SCALE_OUT] = {
-		HLS3_FULL_OAM_3PORTS_SCALE_OUT_SERDES_TYPE,
-		HLS3_FULL_OAM_3PORTS_SCALE_OUT_SERDES_LINK_ID,
-		{0x000000000000, 0x000000000000, 0x000000000000, 0x0100E0000000,
-		 0x000000000000, 0x000000008000, 0x00000000D000, 0x000000000000},
-		{0x00000030F010, 0x0C0000000003, 0x0CF20000000F, 0x000000000300,
-		 0x000000001060, 0x00F001F00000, 0x030000F00000, 0x000000000000},
-		{0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF,
-		 0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF},
-		{0x320000, 0x000320, 0x000320, 0x320000,
-		 0x08C000, 0x00008C, 0x00008C, 0x08C000},
-		{0x3FFF, 0xFFF003, 0xFFF003, 0x3FFF,
-		 0xC00FFF, 0xFFFC00, 0xFFFC00, 0xC00FFF},
-		{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
-	},
-	[GAUDI3_FULLOAM_6PORTS_SCALE_OUT] = {
-		HLS3_FULL_OAM_6PORTS_SCALE_OUT_SERDES_TYPE,
-		HLS3_FULL_OAM_6PORTS_SCALE_OUT_SERDES_LINK_ID,
-		{0x000000000000, 0x000000000000, 0x000000000000, 0x0100E0000000,
-		 0x000000000000, 0x000000008000, 0x00000000D000, 0x000000000000},
-		{0x00000030F010, 0x0C0000000003, 0x0CF20000000F, 0x000000000300,
-		 0x000000001060, 0x00F001F00000, 0x030000F00000, 0x000000000000},
-		{0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF,
-		 0xFFFFFF, 0xFFFFFF, 0xFFFFFF, 0xFFFFFF},
-		{0x374000, 0x000374, 0x000374, 0x374000,
-		 0x1DC000, 0x0001DC, 0x0001DC, 0x1DC000},
-		{0x3FFF, 0xFFF003, 0xFFF003, 0x3FFF,
-		 0xC00FFF, 0xFFFC00, 0xFFFC00, 0xC00FFF},
-		{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
-	},
-	[GAUDI3_SINGLEPORT_OAM_FS_OUT] = {
-		HLS3_SINGLEPORT_OAM_FULLSCALE_OUT_SERDES_TYPE,
-		HLS3_SINGLEPORT_OAM_FULLSCALE_OUT_SERDES_LINK_ID,
-		{0x000000000000, 0x000000000000, 0x000000000000, 0x0100E0000000,
-		 0x000000000000, 0x000000008000, 0x00000000D000, 0x000000000000},
-		{0x00000030F010, 0x0C0000000003, 0x0CF20000000F, 0x000000000300,
-		 0x000000001060, 0x00F001F00000, 0x030000F00000, 0x000000000000},
-		{0x321555, 0x555321, 0x555321, 0x321555,
-		 0x48C555, 0x55548C, 0x55548C, 0x48C555},
-		{0x320000, 0x000320, 0x000320, 0x320000,
-		 0x08C000, 0x00008C, 0x00008C, 0x08C000},
-		{0x3FFF, 0xFFF003, 0xFFF003, 0x3FFF,
-		 0xC00FFF, 0xFFFC00, 0xFFFC00, 0xC00FFF},
-		{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
-	},
-	[GAUDI3_HLB325_FULL_OAM_3PORTS_SCALE_OUT] = {
-		HLB325_FULL_OAM_3PORTS_SCALE_OUT_SERDES_TYPE,
-		HLB325_FULL_OAM_3PORTS_SCALE_OUT_SERDES_LINK_ID,
-		{0x0000F0001030, 0x00002000030F, 0x06000000011F, 0x000200000100,
-		 0x0803000000C0, 0x013F0F303000, 0x0F80C0F01500, 0x0C0104000003},
-		{0x000000000000, 0x000000000000, 0x000000000000, 0x000000000000,
-		 0x000000000000, 0x000000000000, 0x000000000000, 0x000000000000},
-		{0XFFFFFFFF, 0XFFFFFFFF, 0XFFFFFFFF, 0XFFFFFFFF,
-		 0XFFFFFFFF, 0XFFFFFFFF, 0XFFFFFFFF, 0XFFFFFFFF},
-		{0x320000, 0x000320, 0x000320, 0x320000,
-		 0x08C000, 0x00008C, 0x00008C, 0x08C000},
-		{0xCDFFFF, 0xFFFCDF, 0xFFFCDF, 0xCDFFFF,
-		 0xF73FFF, 0xFFFF73, 0xFFFF73, 0xF73FFF},
-		{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
-	},
-};
-
-int hl_serdes_get_serdes_info_core_pcie(const struct device *dev, struct cpucp_nic_info *nic_info,
-	uint16_t serdes_type)
-{
-	int i;
-
-	/* serdes type */
-	nic_info->serdes_type = cfg_pcie.serdes_type;
-
-	/* port enable mask */
-	nic_info->link_mask[0] = cfg_pcie.link_mask;
-
-	/* external ports mask */
-	nic_info->link_ext_mask[0] = cfg_pcie.link_ext_mask;
-
-	/* serdes lane polarity inversion mask */
-	nic_info->pol_tx_mask[0] = cfg_pcie.pol_tx_mask;
-	nic_info->pol_rx_mask[0] = cfg_pcie.pol_rx_mask;
-
-	/* port auto-negotiation enable mask */
-	nic_info->auto_neg_mask[0] = cfg_pcie.auto_neg_mask;
-
-	/* serdes lane swapping map */
-	for (i = 0; i < SERDES_NUM_OF_PHYS; i++)
-		nic_info->tx_swap_map[i] = 0;
-
-	return 0;
-}
-
-int hl_serdes_get_serdes_table(struct hl_serdes_config **tbl, int *len)
-{
-	*tbl = serdes_table;
-	*len = ARRAY_SIZE(serdes_table);
-
-	return 0;
-}
-
-int hl_serdes_get_serdes_info(const struct device *dev, struct cpucp_nic_info *nic_info)
-{
-	return hl_serdes_get_serdes_info_core(dev, nic_info, serdes_type);
-}
-
-static void hl_serdes_init_serdes_type(void)
-{
-	if (hl_is_card_type_pcie())
-		serdes_type = HL338_SERDES_TYPE;
-	else
-		serdes_type = hl_serdes_get_serdes_type();
-
-	cfg_pcie.serdes_type = serdes_type;
-}
-
-uint32_t hl_serdes_get_link_mask(void)
-{
-	struct hl_serdes_config *serdes_table;
-	uint32_t tbl_len;
-	uint16_t serdes;
-	uint8_t module_id;
-
-	if (hl_is_card_type_pcie())
-		return cfg_pcie.link_mask;
-
-	module_id = get_module_id();
-	serdes = hl_serdes_get_serdes_type();
-	hl_serdes_get_serdes_table(&serdes_table, &tbl_len);
-
-	return serdes_table[serdes].enable_mask[module_id];
-}
-
-#if defined(TARGET_IS_PALAD)
-uint16_t hl_serdes_get_serdes_type(void)
-{
-	return GAUDI3_FULLOAM_3PORTS_SCALE_OUT;
-}
-#else
-uint16_t hl_serdes_get_serdes_type(void)
-{
-	uint8_t cpld_link_conf;
-	uint32_t serdes_type;
-	struct hl_serdes_config *table = NULL;
-	int serdes_tbl_len, ret;
-
-	if (dev_serdes_type != HL_SERDES_UNINITIALIZED_TYPE)
-		return dev_serdes_type;
-
-	cpld_link_conf = cpld_get_link_config();
-
-	ret = hl_serdes_get_serdes_table(&table, &serdes_tbl_len);
-	for (serdes_type = 0; serdes_type < serdes_tbl_len; serdes_type++) {
-		if (cpld_link_conf == table[serdes_type].link_id)
-			goto serdes_config;
-	}
-
-	serdes_type = HL_UNKNOWN_SERDES_TYPE;
-	LOG_ERR("Read an unknown serdes type from CPLD 0x%x", cpld_link_conf);
-
-serdes_config:
-	dev_serdes_type = serdes_type;
-
-	return dev_serdes_type;
-}
-#endif
-
-int hl_serdes_get_serdes_info_from_table(const struct device *dev, struct cpucp_nic_info *nic_info,
-	int serdes_type, uint8_t module_id)
-{
-	int ret;
-	unsigned int i, serdes_tbl_len;
-	uint64_t tx_mask, rx_mask;
-	struct hl_serdes_config *serdes_table = NULL;
-
-	hl_serdes_get_serdes_table(&serdes_table, &serdes_tbl_len);
-
-	nic_info->serdes_type = serdes_table[serdes_type].type;
-
-	/* port enable mask */
-	nic_info->link_mask[0] = serdes_table[serdes_type].enable_mask[module_id];
-
-	/* external ports mask */
-	nic_info->link_ext_mask[0] = serdes_table[serdes_type].external_mask[module_id];
-
-	/* serdes lane polarity inversion mask */
-	ret = hl_serdes_get_polarity_mask(&tx_mask, &rx_mask);
-	if (ret) {
-		LOG_ERR("Missing polarity mask");
-		return 1;
-	}
-
-	nic_info->pol_tx_mask[0] = tx_mask ^
-				serdes_table[serdes_type].tx_pol[module_id];
-	nic_info->pol_rx_mask[0] = rx_mask ^
-				serdes_table[serdes_type].rx_pol[module_id];
-
-	/* port autonegotiation enable mask */
-	nic_info->auto_neg_mask[0] = serdes_table[serdes_type].autoneg_mask[module_id];
-
-	/* serdes lane swapping map */
-	for (i = 0; i < SERDES_NUM_OF_PHYS; i++)
-		nic_info->tx_swap_map[i] = serdes_table[serdes_type].tx_lane_swap[i];
-
-	return 0;
-}
-
-int hl_serdes_get_serdes_info_core_mezz(const struct device *dev, struct cpucp_nic_info *nic_info,
-				   int serdes_type)
-{
-	int ret;
-	uint8_t module_id = get_module_id();
-
-	ret = hl_serdes_get_serdes_info_from_table(dev, nic_info, serdes_type, module_id);
-	if (ret)
-		return ret;
-
-	LOG_INF("serdes_type: %d", nic_info->serdes_type);
-	LOG_INF("nic_link_mask: 0x%llx", nic_info->link_mask[0]);
-	LOG_INF("nic_link_ext_mask:0x%llx", nic_info->link_ext_mask[0]);
-	LOG_INF("nic_pol_tx_mask:0x%llx", nic_info->pol_tx_mask[0]);
-	LOG_INF("nic_pol_rx_mask:0x%llx", nic_info->pol_rx_mask[0]);
-	LOG_INF("nic_auto_neg_mask:0x%llx", nic_info->auto_neg_mask[0]);
-
-	return 0;
-}
-
-int hl_serdes_get_serdes_info_core(const struct device *dev, struct cpucp_nic_info *nic_info,
-					uint16_t serdes_type)
-{
-	int ret;
-
-	if (hl_is_card_type_pcie())
-		ret = hl_serdes_get_serdes_info_core_pcie(dev, nic_info, serdes_type);
-	else
-		ret = hl_serdes_get_serdes_info_core_mezz(dev, nic_info, serdes_type);
-
-	if (!ret)
-		/* apply optional boot mask */
-		nic_info->link_mask[0] &= get_boot_nic_mask();
-
-	return ret;
-}
-
-#else
 static struct hl_serdes_module_config module_config;
 static struct hl_serdes_link_id_config link_config;
 static struct hl_serdes_rack_data rack_data = {
@@ -578,33 +264,6 @@
 	module_config.type = HL338_SERDES_TYPE;
 }
 
-#endif
-
-#ifdef SERDES_TABLE_SUPPORT
-static void hl_serdes_init_link_config(void)
-{
-	int err;
-	uint8_t alert_status8;
-
-	if (!hl_is_card_type_pcie())
-		return;
-
-	err = cpld_get_register(CPLD_ALERT_STATUS8, &alert_status8);
-	if (err)
-		alert_status8 = HL338_BRIDGE_NOT_CONNECTED;
-	else
-		alert_status8 &= HL338_BRIDGE_PRESENT_MASK;
-
-	if (alert_status8 == HL338_BRIDGE_PRESENT) {
-		err = cpld_get_link_config_may_fail(&(cfg_pcie.link_config));
-		if (err)
-			cfg_pcie.link_config = HL338_BRIDGE_NOT_CONNECTED;
-		else
-			cfg_pcie.link_config &= HL338_BRIDGE_LINK_CONFIG_MASK;
-	} else
-		cfg_pcie.link_config = HL338_BRIDGE_NOT_CONNECTED;
-}
-#else /* SERDES_TABLE_SUPPORT */
 static int hl_serdes_pci_get_link_config(void)
 {
 	uint8_t link_config;
@@ -628,41 +287,7 @@
 
 	return link_config;
 }
-#endif /* SERDES_TABLE_SUPPORT */
 
-#ifdef SERDES_TABLE_SUPPORT
-static void hl_serdes_init_external_mask(void)
-{
-	const struct device *qsfp1_dev = QSFP1_DEV_INST;
-	const struct device *qsfp2_dev = QSFP2_DEV_INST;
-
-	if (!hl_is_card_type_pcie())
-		return;
-
-	cfg_pcie.link_ext_mask = 0;
-
-	if (!qsfp1_dev) {
-		LOG_ERR("QSFP1 device is missing\n");
-		return;
-	}
-	if (!qsfp2_dev) {
-		LOG_ERR("QSFP2 device is missing\n");
-		return;
-	}
-
-	/*
-	 * In case of Loopback (standalone)
-	 * all the ports are external and enabled.
-	 * QSFP conditioned by detection.
-	 */
-	cfg_pcie.link_ext_mask =
-		(cfg_pcie.link_config != HL338_BRIDGE_LPBK) ? 0 : HL338_MASK_SA_EXT_LPBK;
-	if (hl_sfp_dev_is_enabled(qsfp1_dev))
-		cfg_pcie.link_ext_mask |= QSFP1_ENABLE_MASK;
-	if (hl_sfp_dev_is_enabled(qsfp2_dev))
-		cfg_pcie.link_ext_mask |= QSFP2_ENABLE_MASK;
-}
-#else /* SERDES_TABLE_SUPPORT */
 static void hl_serdes_pci_init_external_mask(int link_config)
 {
 	const struct device *qsfp1_dev = QSFP1_DEV_INST;
@@ -691,17 +316,7 @@
 	if (hl_sfp_dev_is_enabled(qsfp2_dev))
 		module_config.external_mask |= QSFP2_ENABLE_MASK;
 }
-#endif /* SERDES_TABLE_SUPPORT */
 
-#ifdef SERDES_TABLE_SUPPORT
-static void hl_serdes_init_enable_mask_pcie(void)
-{
-	if (cfg_pcie.link_config == HL338_BRIDGE_NOT_CONNECTED)
-		cfg_pcie.link_mask = cfg_pcie.link_ext_mask;
-	else
-		cfg_pcie.link_mask = HL338_EXTERNAL_MASK | cfg_pcie.link_ext_mask;
-}
-#else /* SERDES_TABLE_SUPPORT */
 static void hl_serdes_pci_init_enable_mask(int link_config)
 {
 	if (link_config == HL338_BRIDGE_NOT_CONNECTED)
@@ -709,31 +324,7 @@
 	else
 		module_config.enable_mask = HL338_EXTERNAL_MASK | module_config.external_mask;
 }
-#endif /* SERDES_TABLE_SUPPORT */
 
-#ifdef SERDES_TABLE_SUPPORT
-static void hl_serdes_init_enable_mask(void)
-{
-	if (hl_is_card_type_pcie())
-		hl_serdes_init_enable_mask_pcie();
-}
-#endif /* SERDES_TABLE_SUPPORT */
-
-#ifdef SERDES_TABLE_SUPPORT
-static void hl_serdes_init_polarity_masks(void)
-{
-	if (!hl_is_card_type_pcie())
-		return;
-
-	if (cfg_pcie.link_config != HL338_BRIDGE_LPBK) {
-		cfg_pcie.pol_tx_mask = HL338_TX_POLARITY;
-		cfg_pcie.pol_rx_mask = HL338_RX_POLARITY;
-	} else {
-		cfg_pcie.pol_tx_mask = HL338_TX_POLARITY ^ HL338_TX_POLARITY_SA_EXT_LPBK;
-		cfg_pcie.pol_rx_mask = HL338_RX_POLARITY ^ HL338_RX_POLARITY_SA_EXT_LPBK;
-	}
-}
-#else /* SERDES_TABLE_SUPPORT */
 static void hl_serdes_pci_init_polarity_masks(int link_config)
 {
 	if (link_config != HL338_BRIDGE_LPBK) {
@@ -744,25 +335,7 @@
 		module_config.rx_pol = HL338_RX_POLARITY ^ HL338_RX_POLARITY_SA_EXT_LPBK;
 	}
 }
-#endif /* SERDES_TABLE_SUPPORT */
 
-#ifdef SERDES_TABLE_SUPPORT
-static void hl_serdes_init_auto_neg_mask(void)
-{
-	const struct device *qsfp1_dev = QSFP1_DEV_INST;
-	const struct device *qsfp2_dev = QSFP2_DEV_INST;
-
-	if (!hl_is_card_type_pcie())
-		return;
-
-	cfg_pcie.auto_neg_mask = HL338_AUTONEG_MASK;
-
-	if (hl_sfp_media_is_passive(qsfp1_dev))
-		cfg_pcie.auto_neg_mask |= QSFP1_AUTONEG_MASK;
-	if (hl_sfp_media_is_passive(qsfp2_dev))
-		cfg_pcie.auto_neg_mask |= QSFP2_AUTONEG_MASK;
-}
-#else /* SERDES_TABLE_SUPPORT */
 static void hl_serdes_pci_init_auto_neg_mask(void)
 {
 	const struct device *qsfp1_dev = QSFP1_DEV_INST;
@@ -775,9 +348,7 @@
 	if (hl_sfp_media_is_passive(qsfp2_dev))
 		module_config.autoneg_mask |= QSFP2_AUTONEG_MASK;
 }
-#endif /* SERDES_TABLE_SUPPORT */
 
-#ifndef SERDES_TABLE_SUPPORT
 static void hl_open_rack_eeprom(void)
 {
 	int rc;
@@ -798,9 +369,6 @@
 		break;
 	}
 }
-#else
-static void hl_open_rack_eeprom(void) {}
-#endif
 
 /*
  * The functions invoking order below is important
@@ -810,24 +378,6 @@
  */
 void hl_serdes_serdes_init(void)
 {
-#ifdef SERDES_TABLE_SUPPORT
-	hl_serdes_init_serdes_type();
-
-	hl_serdes_init_link_config();
-	hl_serdes_init_external_mask();
-	hl_serdes_init_enable_mask();
-	hl_serdes_init_polarity_masks();
-	hl_serdes_init_auto_neg_mask();
-
-	LOG_INF("--------");
-	LOG_INF("serdes_type: %d", cfg_pcie.serdes_type);
-	LOG_INF("link_config: %d", cfg_pcie.link_config);
-	LOG_INF("link_ext_mask: 0x%x", cfg_pcie.link_ext_mask);
-	LOG_INF("link_mask: 0x%x", cfg_pcie.link_mask);
-	LOG_INF("pol_tx_mask:0x%llx", cfg_pcie.pol_tx_mask);
-	LOG_INF("pol_rx_mask:0x%llx", cfg_pcie.pol_rx_mask);
-	LOG_INF("auto_neg_mask:0x%x", cfg_pcie.auto_neg_mask);
-#else
 	int link_config = -1;
 
 	if (hl_is_card_type_pcie()) {
@@ -848,7 +398,6 @@
 	LOG_INF("pol_tx_mask:0x%llx", module_config.tx_pol);
 	LOG_INF("pol_rx_mask:0x%llx", module_config.rx_pol);
 	LOG_INF("auto_neg_mask:0x%x", module_config.autoneg_mask);
-#endif
 
 	hl_open_rack_eeprom();
 }
